src/builder/specification.md

```diff
@@ -0,0 +1,339 @@
+# Builder API
+
+This document specifies the Builder API methods that the Consensus Layer uses to interact with external block builders.
+
+```mermaid
+sequenceDiagram
+    participant consensus
+    participant mev_boost
```

Comment:
Should we explicitly mention `mev_boost` here? I wonder if the mermaid diagram is even necessary, given the other specs don't have it

Reply:
could call this "builder software" or something like this

I do think it is worth removing mention to `mev_boost` unless we want to enshrine that name

Reply:
Removed the diagram.